prism/app.py

```diff
@@ -52,21 +52,30 @@ def dispatch_request(self, request):
         except HTTPException as e:
             return e
 
+    def get_customer(self, request):
+        subdomain = request.args.get('customer', None)
+        if subdomain is None:
+            subdomain = request.host.split('.' + settings.DOMAIN)[0]
+        with sentry_sdk.configure_scope() as scope:
+            scope.set_tag("customer", subdomain)
+        customer = self.credentials_store.get_customer(subdomain)
+        return customer
+
     def main(self, request):
         path = request.path[1:]
         _, extension = os.path.splitext(path.lower())
-        if extension not in ('.png', '.jpg', '.jpeg'):
+        if extension not in ('.png', '.jpg', '.jpeg', '.gif'):
             raise NotFound()
+        if extension == '.gif' and request.args.get('out', 'gif') == 'gif':
+            customer = self.get_customer(request)
+            return redirect(core.get_s3_url(customer.read_bucket_name, customer.read_bucket_region, path))
```

Comment:
Does this need to be up here? Can it be below, just above ` if args['command'] == 'info':`?

Reply:
Yeah, that was the first place I put it, but here `args = parse_args(path, request)` function does some validations like `width or height is required`. For **basic** `.gif` case we don't need that.

If the duplicate declaration of `get_customer()` method is the concern, we may call it before the `if extension == '.gif' and request.args.get('out', 'gif') == 'gif':` line. But, here I just don't wanted to break the existing flow of calling the `get_customer` method if all validations are done successfully.

Reply:
> validations like width or height is required. For basic .gif case we don't need that.
We don't actually want to change this behavior. At least not for the gif case specifically. We just want gifs to work with the same parameters as passed for any other image. Right now a width or height is required so the same should be true when serving gifs, even if they do nothing.

Reply:
Ok, right, I will move it as you suggested.